Performance-friendly changing of a Sprites opacity

dnbow opened the pull request:
I dont know if the fading fog is still wanted as it was a place holder so I removed it

dnbow committed:
Performance-friendly changing of a Sprites opacity

I dont know if the fading fog is still wanted as it was a place holder so I removed it

lillithrosepup commented:
swag

keyraven reviewed (changes requested):
This was my original method, however, it does not work properly for pixels that are already partially transparent. set_alpha() sets every pixel to the same value, which is not the intended effect.  The Starclan lineart has a lot of partial transparency.  

I also attempted to leave the lineart unchanged and only apply transparency to the color of the cat, but that effect was deemed too subtle, since it often just looks like the cat was turning black.

dnbow committed:
Update utility.py

havent tested it but this is how I stopped the lineart from turning black (and I think) the cat.

dnbow committed:
fixing oopsies

dnbow commented:
i think this is how i got around that

dnbow committed:
fixed one more oopsie

keyraven commented on scripts/utility.py:
```diff
@@ -1420,6 +1419,17 @@ def update_sprite(cat):
                 temp.blit(new_sprite, (0, 0))
                 new_sprite = temp
+                
+        if cat.opacity < 100 and not cat.prevent_fading and game.settings["fading"]
```
Missing a `:`

keyraven commented on scripts/utility.py:
```diff
@@ -1420,6 +1419,17 @@ def update_sprite(cat):
+                
+        if cat.opacity < 100 and not cat.prevent_fading and game.settings["fading"]
+            new_sprite.set_alpha(int(cat.opacity * 2.55))
+            faded_sprite = Surface((spriteSize, spriteSize), pygame.HWSURFACE | pygame.SRCALPHA).convert_alpha()
```
Surface is not defined. use `pygame.Surface`

dnbow committed:
last oopsie i swear

keyraven reviewed (changes requested):
Concerns:

- You took the method of not applying the opacity to the lineart.  As I said in my previous review, this method was not used because it was too subtle.  The fading effect must be clear from the list screen, not just the profile screen.  It works well enough for ginger cats, but it not clear enough at the list-page level for black and gray cats.  This is not a method I want to use. 
- You are applying the lineart twice. This doubles the partial transparency for all cats with opacity applied. 
- You are applying the lineart after the masks for missing limbs are applied. This means that when cat starts fading, they gain transparent limbs.

dnbow commented:
ahhhh i see, ill close it for now but if i find a solution that solves that ill reopen it
